org.mwc.debrief.limpet_integration/src/org/mwc/debrief/limpet_integration/measured_data/MeasuredDataOperations.java

```diff
@@ -0,0 +1,204 @@
+import org.eclipse.core.runtime.IAdaptable;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.jface.action.Action;
+import org.eclipse.jface.action.IAction;
+import org.eclipse.jface.action.IMenuManager;
+import org.eclipse.jface.action.Separator;
+import org.mwc.cmap.core.CorePlugin;
+import org.mwc.cmap.core.operations.CMAPOperation;
+import org.mwc.cmap.core.property_support.RightClickSupport.RightClickContextItemGenerator;
+import org.mwc.debrief.core.providers.measured_data.DatasetWrapper;
+
+import Debrief.Wrappers.Extensions.Measurements.DataFolder;
+import Debrief.Wrappers.Extensions.Measurements.TimeSeriesCore;
+import Debrief.Wrappers.Extensions.Measurements.TimeSeriesDatasetDouble;
+import MWC.GUI.Editable;
+import MWC.GUI.Layer;
+import MWC.GUI.Layers;
+
+public class MeasuredDataOperations implements RightClickContextItemGenerator
+{
+
+  protected interface CalcOperation
+  {
+    /**
+     * perform a calculation
+     * 
+     * @param items
+     * @return
+     */
+    TimeSeriesCore calculate(List<TimeSeriesDatasetDouble> items);
+  }
+
+  protected static class DatasetsOperation extends CMAPOperation
+  {
+
+    final private CalcOperation _operation;
+    final private List<TimeSeriesDatasetDouble> _items;
+    private TimeSeriesCore _newData;
+    private DataFolder _target;
+
+    public DatasetsOperation(String title, CalcOperation operation,
+        List<TimeSeriesDatasetDouble> fWrappers)
+    {
+      super(title);
+      _operation = operation;
+      _items = fWrappers;
+    }
+
+    @Override
+    public IStatus execute(IProgressMonitor monitor, IAdaptable info)
+        throws ExecutionException
+    {
+      // sort out the destination
+      _target = getTarget();
+
+      // calculate the dataset
+      _newData = _operation.calculate(_items);
+
+      // and store it
+      _target.add(_newData);
+
+      return Status.OK_STATUS;
+    }
+
+    private DataFolder getTarget()
+    {
+      // look at the first item
+      TimeSeriesCore first = _items.get(0);
+
+      DataFolder target = first.getParent();
+
+      return target;
+    }
+
+    @Override
+    public IStatus undo(IProgressMonitor monitor, IAdaptable info)
+        throws ExecutionException
+    {
+      // ok, delete the dataset
+      _target.remove(_newData);
+
+      return Status.OK_STATUS;
+    }
+
+    @Override
+    public IStatus redo(IProgressMonitor monitor, IAdaptable info)
+        throws ExecutionException
+    {
+      // ok, put the dataset back into the parent
+      _target.add(_newData);
+
+      return Status.OK_STATUS;
+    }
+  }
+
+  private class DoAdd implements CalcOperation
+  {
+
+    @Override
+    public TimeSeriesCore calculate(List<TimeSeriesDatasetDouble> items)
+    {
+      long[] times = new long[]
+      {1000, 2000, 3000};
+      double[] values = new double[]
+      {44, 55, 66};
+      
```

Comment:
@theanuradha - I've got stuck on the following line. This plugin can't see **Dataset**.  Dataset is in the **january.jar** that is shared by debrief.legacy